src/pu-utils.c

```diff
@@ -139,6 +150,30 @@ pu_make_filesystem(const gchar *part,
     return TRUE;
 }
 
+gboolean
+pu_set_ext_label(const gchar *part,
+                 const gchar *label,
+                 GError **error)
+{
+    g_autofree gchar *cmd = NULL;
+
+    g_return_val_if_fail(part != NULL, FALSE);
+    g_return_val_if_fail(label != NULL, FALSE);
+    g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+    if (g_strcmp0(label, "") == 0)
+        return TRUE;
```

Comment:
In this case the information on the disk is from the raw ext4 file. Setting an empty label will overwrite that.
This has nothing to do with the partition table, but that is something I'll check.

Reply:
I think it is a valid use-case to enter an empty label to overwrite any existing information. Maybe remove the null-check for label above and do the following:

* if label is null, do nothing and return
* treat empty string the same way as a regular label

I am not sure if `e2label` likes setting empty labels though. That would need testing.

Reply:
Your suggestion works and setting an empty label with `e2label` is also no problem.
The only thing we have to think about is the default value of `label` in case it is not provided.